### PR TITLE
Fit and Finish Changes for DataStreams and Rollups

### PR DESCRIPTION
**Pull request by skumawat2025**
### Description
Making fit and finish changes for DataStreams, Rollups and Notification settings pages according to the [guidance](https://quip-amazon.com/gyAxAkmHcqMb/Trineo-Fit-and-Finish-Guidance)

The change mostly include.
1. Setting up the correct text size of heading according to guide.
2. Using EuiPanel for the even padding across content.

## Video: 
https://github.com/user-attachments/assets/c5c6cdb2-bdf2-4c01-b423-34f8edef3a06

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by RamakrishnaChilaka** [comment]
@skumawat2025 can you please add details to the PR description.

File: public/pages/CreateDataStream/containers/DataStreamDetail/DataStreamDetail.tsx
Comment (danieldong51) [inline public/pages/CreateDataStream/containers/DataStreamDetail/DataStreamDetail.tsx:1]: Should the title of this page be changed to be `<h1>` wrapped in `<EuiText size="s">`? (Guidance 1: Header sizes)

**Comment by danieldong51** [comment]
Not sure if in scope, but the context menu here (`public/pages/DataStreams/containers/DataStreamsActions/index.tsx`) should have no padding:
Data streams context menu

File: public/pages/DataStreams/containers/DataStreams/DataStreams.tsx
Comment (danieldong51) [inline public/pages/DataStreams/containers/DataStreams/DataStreams.tsx:1]: The empty prompt texts should be `<EuiText size="s">` (currently they are just `<EuiText>`

**Comment by danieldong51** [comment]
The text here should also be wrapped in `<EuiText size="s">` in `public/pages/Rollups/components/RollupEmptyPrompt/RollupEmptyPrompt.tsx`
Rollup jobs empty

**Comment by danieldong51** [comment]
The modal header here (`public/pages/RollupDetails/containers/RollupDetails/RollupDetails.tsx`) should be `<h2>` wrapped in `<EuiText size="s">`:

Screenshot 2024-08-28 at 11 29 15 AM

**Comment by danieldong51** [comment]
The modal text here should be `<EuiText size="s">`:

`public/pages/RollupDetails/containers/RollupDetails/RollupDetails.tsx`
Screenshot 2024-08-28 at 11 31 52 AM

**Comment by RamakrishnaChilaka** [comment]
@danieldong51 Thanks a lot for reviewing this PR. @skumawat2025 let's work with Dan to get his approval on this PR.

File: public/pages/CreateDataStream/containers/DataStreamDetail/DataStreamDetail.tsx
Comment (skumawat2025) [reply public/pages/CreateDataStream/containers/DataStreamDetail/DataStreamDetail.tsx:1]: the heading of details page is handled by header and currently it's se to h1 for new home page. Made this change for the old UI case as well.

File: public/pages/DataStreams/containers/DataStreams/DataStreams.tsx
Comment (skumawat2025) [reply public/pages/DataStreams/containers/DataStreams/DataStreams.tsx:1]: Made this change. Thanks

**Comment by astute-decipher** [comment]
LGTM, reviewed the pages offline.

**Review by SuZhou-Joe — APPROVED** [review]
Will merge it on 08/30 Beijing time if no more feedback.

**Comment by danieldong51** [comment]
I found a few more small changes, but everything else looked good!

Context menu in 
- `public/pages/Rollups/containers/Rollups/Rollups.tsx` should be `size="s"`
Screenshot 2024-08-29 at 12 08 29 PM

The titles when using the old UI for the following pages should be `<EuiText size="s"> <h1> {title} <h1> <EuiText size="s">` (not sure if these were intentional):
- `public/pages/EditRollup/containers/EditRollup.tsx`
- The four create rollup steps in `public/pages/CreateRollup/containers`
- `public/pages/Notifications/containers/Notifications/Notifications.tsx`
- `public/pages/ForceMerge/container/ForceMerge/ForceMerge.tsx`
- `public/pages/Rollover/containers/Rollover/Rollover.tsx`

**Comment by skumawat2025** [comment]
> I found a few more small changes, but everything else looked good!
> 
> Context menu in
> 
> * `public/pages/Rollups/containers/Rollups/Rollups.tsx` should be `size="s"`
> 
> Screenshot 2024-08-29 at 12 08 29 PM
> The titles when using the old UI for the following pages should be `<EuiText size="s"> <h1> {title} <h1> <EuiText size="s">` (not sure if these were intentional):
> 
> * `public/pages/EditRollup/containers/EditRollup.tsx`
> * The four create rollup steps in `public/pages/CreateRollup/containers`
> * `public/pages/Notifications/containers/Notifications/Notifications.tsx`
> * `public/pages/ForceMerge/container/ForceMerge/ForceMerge.tsx`
> * `public/pages/Rollover/containers/Rollover/Rollover.tsx`

Thanks for looking into these. Have made these above changes in the latest commit.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**
